### DISCUSSION TITLE
Where does the CRD image/icon come from?

### DISCUSSION THREAD

**Discussion by mendy-lp** [aptakube/aptakube#274] [Q&A] [answered]
As seen in this image 
![image](https://github.com/user-attachments/assets/17fd8023-2177-4f10-84ff-5201812ae0ee)
CRDS have an icon next to them, I am wondering where does this image url come from?

Is it part of the crds?
Is it something that aptkube adds?
Is there a way to inspect which image urls are being loaded?
Is there a way to disable such image loading?

**Comment by goenning** [comment, marked as answer]
Hi @mendy-lp 

1. I am wondering where does this image url come from?

These images come from our servers, example: `https://aptakube.com/favicons?d=cilium.io` which is just a proxy to get favicon from `https://cilium.io`

2. Is it part of the crds?

No

3. Is it something that aptkube adds?

Yes

4. Is there a way to inspect which image urls are being loaded?

You should be able using a Proxy like (https://proxyman.io/), but it's always `https://aptakube.com/favicons?d=cilium.io` and it falls back to a static "Plug" icon.

5. Is there a way to disable such image loading?

Not currently, but I could add a setting for that.

---

Are you concerned about something?

**Reply by goenning** [reply]
after replying to this I just realised I could ship icons for the most popular CRDs (like the ones you shared above) directly inside the app, and only use the server as a fallback, and then the Plug icon as a second fallback 🤔

**Reply by mendy-lp** [reply]
> I could ship icons for the most popular CRDs (like the ones you shared above) directly inside the app, 

That sounds like a great idea, I like it!

I would also like to disable remote icon loading if possible.
I don't like the idea that aptkube will be pinging a remote server for every new crd group it finds

**Reply by goenning** [reply]
this will be available on 1.7.1 :)

![Screenshot 2024-08-22 at 11 31 31](https://github.com/user-attachments/assets/747511e2-3efa-4896-84bc-d5484f7bb0f3)
